### ISSUE TITLE
Request Data working?

### ISSUE THREAD

**Issue by jenkink** [apigee-127/swagger-test-templates#150]
Hi ive been trying to use the request Data parameter but to no avail following the documentation

```
 requestData: {
      '/auth/callback': {
        get: {
          200: [{ code: "test-code", state: "test-state", description: 'some description' }]
        }
      }
```

however i still end up with 
```
qs: {
          code: 'DATA GOES HERE',state: 'DATA GOES HERE'
        },
```

in the request in the generated test

heres the parameters set in the route 

```
"parameters": [
          {
            "name": "code",
            "in": "query",
            "required": true,
            "type": "string",
            "description": "OAuth2 response code"
          },
          {
            "name": "state",
            "in": "query",
            "required": true,
            "type": "string",
            "description": "State send back from OAuth our session SessionId"
          }
        ],
```
PR #137 seems to fix this issue but im still not getting data

am i missing something obvious?

**Comment by baynezy** [comment]
@jenkink can you post the swagger file at all?

**Comment by jenkink** [comment]
is there a way i can send it directly? dont want to post it

**Comment by noahdietz** [comment]
I just injected a query string param successfully with a simple API I have. 

Here is a gist with all of the files: https://gist.github.com/noahdietz/f20f5668038a25099de408de9f31ff15

**Comment by jenkink** [comment]
tried the simple API and had success, so it must be an issue with my swagger then, ive stripped it down to only the route ive been trying to get to work 

```
{
  "swagger": "2.0",
  "info": {
    "version": "0.0.1",
    "title": "test",
    "description": "test"
  },
  "host": "localhost:10010",
  "basePath": "v1",
  "schemes": [
    "https"
  ],
  "consumes": [
    "application/json"
  ],
  "produces": [
    "application/json"
  ],
  "x-amazon-apigateway-request-validators": {
    "all": {
      "validateRequestBody": true,
      "validateRequestParameters": true
    },
    "params-only": {
      "validateRequestBody": false,
      "validateRequestParameters": true
    }
  },
  "paths": {
    "/auth/callback": {
      "get": {
        "description": "Authentication callback endpoint\n",
        "operationId": "authCallback",
        "tags": [
          "Authentication"
        ],
        "parameters": [
          {
            "name": "code",
            "in": "query",
            "required": true,
            "description": "code"
            "type": "string"
          },
          {
            "name": "state",
            "in": "query",
            "required": true,
            "type": "string",
            "description": "State"
          }
        ],
        "responses": {
          "200": {
            "description": "Returns sessionId",
            "schema": {
              "type": "object",
              "items": {
                "$ref": "#/definitions/Session"
              }
            }
          },
          "default": {
            "description": "Unexpected error",
            "schema": {
              "type": "object",
              "items": {
                "$ref": "#/definitions/Error"
              }
            }
          }
        }
      }
    }
  },
  "definitions": {
    "Error": {
      "type": "object",
      "properties": {
        "code": {
          "type": "integer",
          "format": "int32"
        },
        "message": {
          "type": "string"
        },
        "fields": {
          "type": "string"
        }
      }
    },
    "Session": {
      "type": "object",
      "properties": {
        "session-id": {
          "type": "string"
        }
      }
    }
  },
  "parameters": {
    "Session": {
      "name": "session-id",
      "in": "header",
      "required": true,
      "type": "string"
    }
  }
}

```

**Comment by noahdietz** [comment]
I'm going to update here as I debug.

I logged the params to the  `requestDataParamFormatter`helper [here](https://github.com/apigee-127/swagger-test-templates/blob/master/lib/helpers.js#L228) and found that the `requestParameters` object is empty.

**Comment by noahdietz** [comment]
looks like the `config.requestData` parsing code is [using](https://github.com/apigee-127/swagger-test-templates/blob/master/index.js#L204) the `requestPath`, which is prepended with the `swagger.basePath` if such a property exists [(here)](https://github.com/apigee-127/swagger-test-templates/blob/master/index.js#L93), rather than just the path segment defined in the spec that's being evaluated. It's looking for `v1/auth/callback` in the `config.requestData`. 

This is a bug on our part, with an easy work around for the time being. 

@jenkink please use the full path `v1/auth/callback` in your `config.requestData` definitions until we can release a fix.

If this doesn't work, let me know. If it does work, please indicate this and leave the issue open so we can address with a PR. Thanks

**Comment by jenkink** [comment]
@noahdietz yes using the full path does work thanks for the help, i will use this work around

**Comment by noahdietz** [comment]
@jenkink released the fix in v1.4.3

Please follow up if there are any more issues or questions. Thanks for the bug report and for using STT! much appreciated.

**Comment by jenkink** [comment]
great thanks!
